### ISSUE TITLE
Tech docs converts punctuation to HTML entities in IDs

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/tech-docs-gem#183]
## What
If a heading includes for instance an apostrophe it gets converted to its entity reference number in the resulting id.

So for instance 
```
## Turn on 'compatibility mode'
```
results in the URL [`https://frontend.design-system.service.gov.uk/compatibility-mode/#39-compatibility-mode-39`](https://frontend.design-system.service.gov.uk/compatibility-mode/#turn-on-39-compatibility-mode-39) 

and 
```
## Find out which version you're using
```
becomes [`https://frontend.design-system.service.gov.uk/updating-with-npm/#find-out-which-version-you-39-re-using`](https://frontend.design-system.service.gov.uk/updating-with-npm/#find-out-which-version-you-39-re-using)

## Why
The URLs end up looking pretty weird, this might or might not be an issue for users.

**Comment by JC-DSIT** [comment]
I don't think this has been raised by a user and also it's difficult to do much about the encoding so closing for now.  May re-visit in the future
